### PR TITLE
refactor(useWebSocket)!: change immediate default for 5.0.0

### PR DESCRIPTION
**Pull request by Shinigami92**
https://github.com/vueuse/vueuse/issues/501#issuecomment-839726620

### CODE REVIEW

File: packages/core/useWebSocket/index.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ export function useWebSocket<Data = any>(
     heartbeatResume = resume
   }
 
-  if (options.immediate) _init()
+  if (options.immediate === false) _init()
```
Comment (antfu) [inline packages/core/useWebSocket/index.ts:226]: emm?

Comment (Shinigami92) [reply packages/core/useWebSocket/index.ts:226]: Oops, seems it was very late yesterday 🤣
Will change it when I'm out of bed

File: packages/core/useWebSocket/index.ts [hunk 2]
```diff
@@ -223,7 +223,7 @@ export function useWebSocket<Data = any>(
     heartbeatResume = resume
   }
 
-  if (options.immediate) _init()
+  if (options.immediate !== false) _init()
```
Comment (wheatjs) [inline packages/core/useWebSocket/index.ts:226]: could just be `if (options.immediate) _init()` right? 😅

Comment (Shinigami92) [reply packages/core/useWebSocket/index.ts:226]: No, cause the new default is that if it's `undefined` than it should also count as `true`
That's the whole point of this PR 😛

Comment (wheatjs) [reply packages/core/useWebSocket/index.ts:226]: My bad 😅

Comment (antfu) [reply packages/core/useWebSocket/index.ts:226]: Better to assign default value in L121 instead.

**Pull request merged**